host/src/gatt.rs

```diff
@@ -138,11 +142,35 @@ pub enum GattEvent<'reference> {
     },
 }
 
-pub struct GattClient<'reference, 'resources, T: Controller, const MAX: usize, const L2CAP_MTU: usize = 27> {
-    pub(crate) services: Vec<ServiceHandle, MAX>,
+pub struct NotificationListener<'t, const MAX_NOTIF: usize, const NOTIF_QSIZE: usize, const ATT_MTU: usize> {
+    pub(crate) listener: DynamicReceiver<'t, Vec<u8, ATT_MTU>>,
+}
+
+impl<'t, const MAX_NOTIF: usize, const NOTIF_QSIZE: usize, const ATT_MTU: usize>
+    NotificationListener<'t, MAX_NOTIF, NOTIF_QSIZE, ATT_MTU>
+{
+    #[allow(clippy::should_implement_trait)]
+    pub fn next(&mut self) -> impl Future<Output = Vec<u8, ATT_MTU>> + '_ {
+        self.listener.receive()
+    }
+}
+
+pub struct GattClient<
+    'reference,
+    'resources,
+    T: Controller,
+    const MAX_SERVICES: usize,
+    const MAX_NOTIF: usize,
+    const NOTIF_QSIZE: usize,
+    const L2CAP_MTU: usize = 27,
+    const ATT_MTU: usize = 23,
```

Comment:
These are a bit hard to keep in sync. I'd rather just use l2cap_mtu for the vec size to avoid them.